servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HeaderUtils.java

```diff
@@ -112,8 +102,8 @@ static boolean canAddResponseTransferEncodingProtocol(final int statusCode,
     }
 
     private static boolean canAddContentLength(final HttpMetaData metadata) {
-        return !hasContentHeaders(metadata.headers()) &&
-                isSafeToAggregate(metadata) && !mayHaveTrailers(metadata);
+        return !hasContentHeaders(metadata.headers()) && isSafeToAggregate(metadata) &&
+                (!mayHaveTrailers(metadata) || metadata.version().major() > 1);
```

Comment:
nit: consider swapping order of `mayHaveTrailers` and `metadata.version().major() > 1` -> short circuit evaluation may avoid casting if h2+